include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+#define FDCAN_TXELEM_ID_FLAG_RTR BIT29
+/** @{ Mask and shift for the frame message id */
+#define FDCAN_TXELEM_ID_29_MASK 0x1FFFFFFF
+#define FDCAN_TXELEM_ID_29_SHIFT 0
+#define FDCAN_TXELEM_ID_11_MASK 0x7FF
+#define FDCAN_TXELEM_ID_11_SHIFT 18
+/** @} */
+
+/** @{ Message marker (arbitrary user tag), written by CPU during Tx buffer
+ * configuration. Copied into Tx event FIFO element for identification of TX
+ * message. */
+#define FDCAN_TXELEM_MSG_MM_MASK 0xFF
+#define FDCAN_TXELEM_MSG_MM_SHIFT 24
+/** @} */
+/** Event FIFO control, if set, on transmission, event will be stored into the
+ * event FIFO */
+#define FDCAN_TXELEM_MSG_EFC BIT23
+/** FD Format, if set message may be longer than 8 bytes and may switch baud
+ * rates */
+#define FDCAN_TXELEM_MSG_FDF BIT21
+/** Baud Rate Switch, if set message will switch baud rates during data
+ * transmission */
+#define FDCAN_TXELEM_MSG_BRS BIT20
+/** @{ Message data length code */
+#define FDCAN_TXELEM_MSG_DLC_MASK 0xF
+#define FDCAN_TXELEM_MSG_DLC_SHIFT 16
+/** @} */
+
+/* --- RX Event Definition -------------------------------------------------- */
+/** Element written to the TX Event FIFO on transmission event */
+struct fdcan_txevent {
+	uint32_t id_flag;
+	uint32_t msg_flag;
+};
+
+/** Error status indicator; when clear the ESI bit in the transmitted message
+ * depends on the controllers error passive flag. If set, the ESI bit is
+ * transmitted recessive. */
+#define FDCAN_TXEVENT_ID_FLAG_ESI BIT31
+/** Extended ID indicator; when set the frame uses 29 bit addressing */
+#define FDCAN_TXEVENT_ID_FLAG_XTD BIT30
+/** Remote transmission requested; when set the frame is a remote frame */
+#define FDCAN_TXEVENT_ID_FLAG_RTR BIT29
+/** @{ Mask and shift for the frame message id */
+#define FDCAN_TXEVENT_ID_29_MASK  0x1FFFFFFF
+#define FDCAN_TXEVENT_ID_29_SHIFT 0
+#define FDCAN_TXEVENT_ID_11_MASK  0x7FF
+#define FDCAN_TXEVENT_ID_11_SHIFT 18
+/** @} */
+
+/** @{ Message marker (arbitrary user tag), copied from the TX element. */
+#define FDCAN_TXEVENT_MSG_MM_MASK 0xFF
+#define FDCAN_TXEVENT_MSG_MM_SHIFT 24
+/** @} */
+/** FD Format, if set message may be longer than 8 bytes and may switch baud
+ * rates */
+#define FDCAN_TXEVENT_MSG_FDF BIT21
+/** Baud Rate Switch, if set message will switch baud rates during data
+ * transmission */
+#define FDCAN_TXEVENT_MSG_BRS BIT20
+/** @{ Message data length code */
+#define FDCAN_TXEVENT_MSG_DLC_MASK  0xF
+#define FDCAN_TXEVENT_MSG_DLC_SHIFT 16
+/** @} */
+/** @{ TX timestamp counter value captured on start of frame reception.
+ * Resolution set by timestamp counter prescaler FDCAN_TSCC.TCP */
+#define FDCAN_TXEVENT_MSG_TIMESTAMP_MASK  0xFFFF
+#define FDCAN_TXEVENT_MSG_TIMESTAMP_SHIFT 0
+/** @} */
+
+/* --- Filter Definition ---------------------------------------------------- */
+/** 11 bit address filter element. Note: depending on filter mode, the 2nd
+ * address field has different meanings. */
+struct fdcan_stdfilt {
+	uint32_t flags;
+};
```

Comment:
This struct doesn't seem to be doing much?

Reply:
Agreed. However, the thought was that it would be clearer to use these as `struct fdcan_stdfilt[]` vs `uint32_t[]`.